cmake/modules/FindNlohmannJson.cmake

```diff
@@ -0,0 +1,41 @@
+#.rst:
```

Comment:
Basically, when I started implementing I overlooked that there was already a json library in kodi. However, multi nested json objects in rapidjson requires a lot of boilerplate code and is not as concise as in nlohmann::json. I can refactor everything to use rapidjson but eventually it will be a lot less readable. Also nlohmann::json is a header only library so it doesn't add that much overhead

Reply:
I guess we'll need to discuss this in the team; as you might know Kodi already has a ton of dependencies and the maintenance of depends etc. does add up in the end, so we try to keep the number of new deps to a minimum

Reply:
I discussed this with @lrusak, as we've both had the chance to add dependencies to Kodi.

Ultimately, the overhead of a new dependency (even a header-only one) leads to dependency proliferation and a nightmare for all packagers in the entire downstream ecosystem. Think LibreELEC, CoreELEC, OSMC, OpenEmbedded, all platforms that don't use our depends system, all needing to do everything required to package and maintain and constantly keep in sync a new dependency.

RapidJSON certainly has some downsides that you've outlined, but it's the JSON dependency we're stuck with until we decide to fully replace it. Perhaps this is motivation to add some convenience functions in our xbmc/utils folder to address the downsides and lead to nicer code in your work here. But I think whatever approach you take, it should use rapidjson.

Reply:
> Ultimately, the overhead of a new dependency (even a header-only one) leads to dependency proliferation and a nightmare for all packagers in the entire downstream ecosystem. Think LibreELEC, CoreELEC, OSMC, OpenEmbedded, all platforms that don't use our depends system, all needing to do everything required to package and maintain and constantly keep in sync a new dependency.

As long as only webos uses nlohmanjson, this is not a valid argument. No distro needs to do anything atm.

Reply:
Works been done now :+1: 

https://github.com/webosbrew/xbmc/pull/3

Reply:
Done